### ISSUE TITLE
improve XlsExporter column limits

### ISSUE THREAD

**Issue by pjfanning** [OpenRefine/OpenRefine#2600]
**Is your feature request related to a problem or area of OpenRefine? Please describe.**

XlsExporter has this limit:

```
                    if (i == 255 && cells.size() > 256) {
                        c.setCellValue("ERROR: TOO MANY COLUMNS");
```

xlsx format allows wider spreadsheets and POI SpreadsheetVersion class can be used to work these out

**Describe the solution you'd like**

* The different POI workbook implementations have a getSpreadsheetVersion which has the limit data in it (older spreadsheet formats have tighter limits) - so code can be customised based on this metadata 
* I can provide a PR if this improvement seems useful. 

**Additional context**
Based on a recent review of the excel code

**Comment by wetneb** [comment]
That sounds very useful, 256 columns is a very low limit.

I know basically nothing about XLS(X), but I assume that this will come to the cost of producing spreadsheets in a more recent format, so perhaps not as widely supported as what we are currently using? Would it make sense to dynamically choose the oldest format that supports a given number of columns?

**Comment by wetneb** [comment]
Closed by #2602
